### ISSUE TITLE
unknown function error

### ISSUE THREAD

**Issue by Mixpap** [robertfeldt/BlackBoxOptim.jl#189]
Hi, I get a huge dump of `unknown function` error when I run an optimization, for example:
```julia
function L(p) 
    return abs(100.0-p[1])
end
opt = bbsetup(L; SearchRange =collect(zip([-200.0],[200.0])),PopulationSize=20,MaxSteps = 100, TraceMode = :compact)
res1 = bboptimize(opt)
```
```
unknown function (ip: 0x7f2127bf18c4)
unknown function (ip: 0x7f2127bf18ff)
unknown function (ip: 0x7f213974354d)
unknown function (ip: 0x7f213977953f)
unknown function (ip: 0x7f21397790fc)
unknown function (ip: 0x7f21397790fc)
jl_toplevel_eval_in at /usr/bin/../lib/julia/libjulia-internal.so.1 (unknown line)
unknown function (ip: 0x7f2127cc99f7)
unknown function (ip: 0x7f2127d21338)
unknown function (ip: 0x7f2127d20db2)
unknown function (ip: 0x7f2127a546cc)
unknown function (ip: 0x7f2127a56dec)
unknown function (ip: 0x7f2127a56f55)
unknown function (ip: 0x7f213979b76e)
repl_entrypoint at /usr/bin/../lib/julia/libjulia-internal.so.1 (unknown line)
main at julia (unknown line)
__libc_start_main at /usr/bin/../lib/libc.so.6 (unknown line)
_start at julia (unknown line)
```
The Project.toml of the project is
```
[a0f608ac] AffineInvariantMCMC v1.0.1
  [c7e460c6] ArgParse v1.1.4
  [a134a8b2] BlackBoxOptim v0.6.0
  [336ed68f] CSV v0.9.4
  [13f3f980] CairoMakie v0.6.5
  [5ae59095] Colors v0.12.8
  [76746363] Cosmology v1.0.0
  [31c24e10] Distributions v0.24.18
  [f6369f11] ForwardDiff v0.10.19
  [e9467ef8] GLMakie v0.4.6
  [916415d5] Images v0.24.1
  [a98d9a8b] Interpolations v0.13.4
  [033835bb] JLD2 v0.4.14
  [6fdf6af0] LogDensityProblems v0.11.0
  [ee78f7c6] Makie v0.15.2
  [2913bbd2] StatsBase v0.33.10
  [84d833dd] TransformVariables v0.4.1
  [3a884ed6] UnPack v1.0.2
  [1986cc42] Unitful v1.9.0
  [6fb2a4bd] UnitfulAngles v0.6.1
  [6112ee07] UnitfulAstro v1.0.1
``` 
and the Julia version is
```
Julia Version 1.6.2
Commit 1b93d53fc4* (2021-07-14 15:36 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-12.0.1 (ORCJIT, skylake)
```
All the tests are passed.

**Comment by ConnectedSystems** [comment]
Hello,

Not affiliated with BlackBoxOptim in any way, but your MWE works on Julia v1.6.3 (on Windows, however).

I wonder if this is an installation issue, or a bug fixed up in 1.6.3. 

I also wonder what this part of the code is for: `collect(zip([-200.0],[200.0]))` as it would resolve to something like `[(-200.0,200.0)]` , but I guess its to reflect something you're doing and the context is just missing here.

**Comment by Mixpap** [comment]
Updated to 1.6.3 and updated the installation and now it seems ok. Thanks.
(Yes, I forgot to change it. I use the optimization to a non-fixed number of parameters, so the limits are updating along the procedure)
